python/ray/_private/node.py

```diff
@@ -382,26 +396,32 @@ def _init_temp(self):
         if self.head:
             self._temp_dir = self._ray_params.temp_dir
         else:
-            temp_dir = ray._private.utils.internal_kv_get_with_retry(
-                self.get_gcs_client(),
-                "temp_dir",
-                ray_constants.KV_NAMESPACE_SESSION,
-                num_retries=NUM_REDIS_GET_RETRIES,
-            )
-            self._temp_dir = ray._private.utils.decode(temp_dir)
+            if self._ray_params.temp_dir is None:
+                temp_dir = ray._private.utils.internal_kv_get_with_retry(
```

Comment:
create a cached wrapper around the internal_kv_get_with_retry?

Reply:
We can, and implementation wise it's not hard (python built in support this).
I think the hard part is **how to handle the stale value**. So far when worker started, we don't have duplicate get. I guess it's ok for now.